homeassistant/components/lutron_caseta/__init__.py

```diff
@@ -94,15 +97,26 @@ async def async_setup_entry(hass, config_entry):
     keyfile = hass.config.path(config_entry.data[CONF_KEYFILE])
     certfile = hass.config.path(config_entry.data[CONF_CERTFILE])
     ca_certs = hass.config.path(config_entry.data[CONF_CA_CERTS])
+    bridge = None
 
-    bridge = Smartbridge.create_tls(
-        hostname=host, keyfile=keyfile, certfile=certfile, ca_certs=ca_certs
-    )
+    try:
+        bridge = Smartbridge.create_tls(
+            hostname=host, keyfile=keyfile, certfile=certfile, ca_certs=ca_certs
+        )
+    except ssl.SSLError:
+        _LOGGER.error("Invalid certificate used to connect to bridge at %s.", host)
+        return False
+
+    timed_out = True
+    try:
+        with async_timeout.timeout(BRIDGE_TIMEOUT):
```

Comment:
I think this should use `async with`.

https://pypi.org/project/async-timeout/

Reply:
Fixed in https://github.com/home-assistant/core/pull/45812